src/kit.ts

```diff
@@ -756,13 +756,18 @@ async function tryCreateNewVCEnvironment(inst: VSInstallation, hostArch: string,
   log.debug(` InstallVersion: ${inst.installationVersion}`);
   if (version) {
     const generatorName: string|undefined = VsGenerators[version[1]];
+    let toolset = undefined;
+    // Visual Studio 2010 doesn't support for choose toolset host such as host=x64
+    if (kit.visualStudio !== "VisualStudio.10.0") {
+      toolset = "host=" + hostArch;
+    }
     if (generatorName) {
       log.debug(` ${localize('generator.present', 'Generator Present: {0}', generatorName)}`);
       kit.preferredGenerator = {
         name: generatorName,
         platform: generatorPlatformFromVSArch[targetArch] as string || targetArch,
         // CMake generator toolsets support also different versions (via -T version=).
-        toolset: "host=" + hostArch
+        toolset
       };
     }
     log.debug(` ${localize('selected.preferred.generator.name', 'Selected Preferred Generator Name: {0} {1}', generatorName, JSON.stringify(kit.preferredGenerator))}`);
```

Comment:
How about we assign toolset as: 
    toolset: "host=" + kitHostTargetArch(hostArch, targetArch, majorVersion < 15)

Reply:
fine with that

Reply:
Great. Can you test that in your setup and if all is well commit an update? I'll approve it.

Reply:
toolset: "host=" + kitHostTargetArch(hostArch, targetArch, majorVersion < 15)

Why host= still here? that's should be removed

Reply:
```
[main] 正在配置文件夹: xemu-with-lib 
[driver] 删除 e:/CI-Cor-Ready/xemu/xemu-with-lib/build-cmake/msvc_10_0_win32_x64_Debug/CMakeCache.txt
[driver] 删除 e:\CI-Cor-Ready\xemu\xemu-with-lib\build-cmake\msvc_10_0_win32_x64_Debug\CMakeFiles
[proc] 执行命令: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -He:/CI-Cor-Ready/xemu/xemu-with-lib -Be:/CI-Cor-Ready/xemu/xemu-with-lib/build-cmake/msvc_10_0_win32_x64_Debug -G "Visual Studio 10 2010" -T host=amd64 -A x64
[cmake] Not searching for unused variables given on the command line.
[cmake] CMake Error at CMakeLists.txt:17 (project):
[cmake]   Generator
[cmake] 
[cmake]     Visual Studio 10 2010
[cmake] 
[cmake]   given toolset specification
[cmake] 
[cmake]     host=amd64
[cmake] 
[cmake]   that contains invalid field 'host=amd64'.
[cmake] 
[cmake] 
[cmake] -- Configuring incomplete, errors occurred!
[cmake] See also "E:/CI-Cor-Ready/xemu/xemu-with-lib/build-cmake/msvc_10_0_win32_x64_Debug/CMakeFiles/CMakeOutput.log".

```

Reply:
Oh, I thought that the problem was using amd64 versus x64, not the host attribute itself. I see now, looks good. I am approving.